descartes_light/solvers/include/descartes_light/solvers/ompl/impl/ompl_solver.hpp

```diff
@@ -0,0 +1,136 @@
+#ifndef DESCARTES_LIGHT_SOLVERS_OMPL_IMPL_OMPL_SOLVER_HPP
+#define DESCARTES_LIGHT_SOLVERS_OMPL_IMPL_OMPL_SOLVER_HPP
+
+#include <descartes_light/solvers/ompl/ompl_solver.h>
+#include <descartes_light/solvers/bgl/impl/event_visitors.hpp>
+#include <descartes_light/solvers/ompl/descartes_space.h>
+
+#include <descartes_light/descartes_macros.h>
+#include <ompl/base/ScopedState.h>
+#include <ompl/geometric/SimpleSetup.h>
+#include <ompl/geometric/planners/rrt/RRTConnect.h>
+#include <ompl/geometric/planners/rrt/RRT.h>
+
+namespace descartes_light
+{
+template <typename FloatType>
+void BGLOMPLSolver<FloatType>::initOMPL()
+{
+  // Convenience aliases
+  auto& graph_ = BGLSolverBase<FloatType>::graph_;
+  const auto& source_ = BGLSolverBase<FloatType>::source_;
+  const auto& ladder_rungs_ = BGLSolverBase<FloatType>::ladder_rungs_;
+  const auto& edge_eval_ = BGLSolverBaseSVDE<FloatType>::edge_eval_;
+
+  auto mod_ladder_rungs = std::move(ladder_rungs_);
+  std::vector<descartes_light::VertexDesc<FloatType>> blank_rung = {source_};
+  mod_ladder_rungs.insert(mod_ladder_rungs.begin(), blank_rung);
+  mod_ladder_rungs.push_back(blank_rung);
+
+
+  dss_ = std::make_shared<descartes_light::DescartesStateSpace<FloatType>>(graph_,
+                                                                           mod_ladder_rungs,
+                                                                           edge_eval_,
+                                                                           max_dist_,
+                                                                           rung_to_rung_dist_);
+
+  dss_->setLongestValidSegmentFraction(0.5);
```

Comment:
Should this be parameterized?

Reply:
So I don't think this actually does anything in the context of how I have the DescartesStateSpace methods defined, but it is required for OMPL. Basically it takes the maximum extent from the state space and multiplies by this factor. Any segment longer than this is checked at intervals equal to this length. We only ever check from point A to point B in Descartes, so I just set this to be a high value. (It also cannot be 1.0 or 0.0)